### PR TITLE
Switch from command line flags to config file based options

### PR DESCRIPTION
**Pull request by shuheiktgw**
## What this PR does / Why we need it
Switched from command line flags to config file-based options. 
https://book.kubebuilder.io/component-config-tutorial/tutorial.html

## Which issue(s) this PR fixes

None

## Steps to test the update

```
make kind-cluster-reset

export IMG=mercari/spanner-autoscaler:local
make docker-build && make kind-load-docker-image

kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.6.1/cert-manager.yaml
make deploy
```

### CODE REVIEW

File: config/default/manager_auth_proxy_patch.yaml [hunk 1]
```diff
@@ -18,9 +18,4 @@ spec:
         - "--v=10"
         ports:
         - containerPort: 8443
-          name: https
-      - name: manager
```
Comment (shuheiktgw) [inline config/default/manager_auth_proxy_patch.yaml:22]: This conflicts with manager_config_patch.yaml so I removed it

File: config/default/manager_auth_proxy_patch.yaml [hunk 2]
```diff
@@ -18,9 +18,4 @@ spec:
...
-        args:
-        - "--health-probe-bind-address=:8081"
-        - "--metrics-bind-address=127.0.0.1:8080"
-        - "--leader-elect"
```
Comment (tkuchiki) [inline config/default/manager_auth_proxy_patch.yaml:26]: `$ kustomize build config/default`

```yaml
...
---
apiVersion: v1
data:
  controller_manager_config.yaml: |
    apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
    kind: ControllerManagerConfig
    health:
      healthProbeBindAddress: :8081
    metrics:
      bindAddress: 127.0.0.1:8080
    webhook:
      port: 9443
    leaderElection:
      leaderElect: true
      resourceName: 54b82eb3.mercari.com
kind: ConfigMap
metadata:
  name: spanner-autoscaler-manager-config
  namespace: spanner-autoscaler
```

**Review by tkuchiki — APPROVED** [review]
LGTM

**Comment by tkuchiki** [comment]
After reviewing @rustycl0ck, we'll merge this PR.

**Review by rustycl0ck — APPROVED** [review]
LGTM

**Pull request merged**